### PR TITLE
User list api bind

### PR DESCRIPTION
**Pull request by harshup20**
created new project integrating the github userlist api

### CODE REVIEW

File: lib/datasource/local/git_user_business_userstore.dart [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import 'dart:async';
+import 'package:flutter_base_architecture/data/local/sharedpreferences/user_stores.dart';
+import 'package:git_users/domain/model/user_domain.dart';
+import 'package:git_users/presentation/model/user_item.dart';
+
+class GitUserUserStore extends UserStore<UserDomain> {
```
Comment (naikdp7) [inline lib/datasource/local/git_user_business_userstore.dart:6]: it should be GitUserStore

File: lib/datasource/remote/providers/rest/response/user_response.dart [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  @override
+  parseResponseData(data, int apiIdentifier) {
+
+    switch (apiIdentifier) {
+      case ApiIdentifier.API_USER_LIST:
+        try {
+          var responseData = data as Map<String, dynamic>;
+          // var list = responseData.getOrElse("results", List()) as List<dynamic>;
+          var list = responseData.getOrElse("", List()) as List<dynamic>;
```
Comment (naikdp7) [inline lib/datasource/remote/providers/rest/response/user_response.dart:22]: empty key parsing?

File: lib/presentation/user_list/user_list_view.dart [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  @override
+  Widget buildBody() {
+    return ScreenTypeLayout(
+      mobile: OrientationLayoutBuilder(
+        portrait: (context) => Container(
+          color: Colors.black,
+          child: Center(
+            child: Text('add the view here'),),),
```
Comment (naikdp7) [inline lib/presentation/user_list/user_list_view.dart:30]: it should be localised

**Review by naikdp7 — CHANGES_REQUESTED** [review]
Requested changes

File: lib/datasource/remote/providers/rest/request/git_user_landing_request.dart [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import 'package:flutter_base_architecture/data/remote/rest_service.dart';
+
+
+abstract class GitUserLandingRestRequest extends RESTRequest {
+  RESTService service;
+ static const int page = 1;
+
+  GitUserLandingRestRequest(this.service,
+      {apiUrl = 'https://api.github.com',
+        schema: "http",
```
Comment (mayyuu) [inline lib/datasource/remote/providers/rest/request/git_user_landing_request.dart:12]: Scheme would be https here

File: lib/datasource/remote/providers/rest/request/user_request.dart [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+class UserRequest extends GitUserLandingRestRequest{
+  UserRequest(RESTService service) : super(service);
+
+  Future getUsers() async {
+    Map<String, dynamic> params = Map();
+    params.putIfAbsent("since", () {
+      return 31;
+    });
+    print('the end point${Endpoint.USERS}');
+    return await execute(Endpoint.USERS.toString(), params,
```
Comment (mayyuu) [inline lib/datasource/remote/providers/rest/request/user_request.dart:17]: Can we remove toString() from here ,there no need to typeCast here

File: lib/presentation/user_list/user_list_view.dart (hunk 3 above)
Comment (harshup20) [reply lib/presentation/user_list/user_list_view.dart:30]: its just for temp. as after this merge request will add the ui

File: lib/datasource/remote/providers/rest/response/user_response.dart (hunk 2 above)
Comment (mayyuu) [inline lib/datasource/remote/providers/rest/response/user_response.dart:22]: Key which will fetch your list is not provided here,please check this

**Review by mayyuu — APPROVED** [review]

File: lib/presentation/user_list/user_list_view.dart [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  @override
+  Widget buildBody() {
+    return ScreenTypeLayout(
+      mobile: OrientationLayoutBuilder(
+        portrait: (context) => Container(
+          color: Colors.black,
+          child: Center(
+            child: Text('add the view here', style: TextStyle(color: Colors.white),),),),
```
Comment (naikdp7) [inline lib/presentation/user_list/user_list_view.dart:30]: it should be localised also please format the code

**Review by naikdp7 — APPROVED** [review]
Approved

**Pull request merged**